### ISSUE TITLE
Seek should bring back range instead of offset

### ISSUE THREAD

**Issue by guypaskar** [gpac/mp4box.js#26]
It would help a lot if seek would return the range of the segment instead of the ony the starting offset incase I want to download only the needed segment.

What do you think?

**Comment by cconcolato** [comment]
I think I understand, maybe I was confused ;) The segment duration (that you can set in the demo app) and the seek are currently not related. You're saying I could relate them to return a range that corresponds to the number of bytes necessary to produce at least one segment (as configured). Interesting. It may not be always possible (e.g. if the input file is fragmented without sidx) but for non-fragmented files this could work. I'll see what I can do.

**Comment by cconcolato** [comment]
Actually, I thought of the seek method for that too. It's separate from the downloader for that purpose. Why don't you want to use two seek operations: one to get the file offset at v.currentTime+20 and one at v.currentTime+20+segmentDuration ? 

Also, be careful when you say "make xhrs that are contains multiple complete segments", this might be misleading. Segments don't exist in the original file (since it's not fragmented), so they won't be in the XHR. In the XHR you will have the samples/frames that will be used to make segments according to the settings you gave to mp4box.js.

**Comment by guypaskar** [comment]
That would work great, but How do I know what is the the segment duration in time? I'm not sure...

Yes, this might be misleading. When I say "make xhrs that are contains multiple complete segments"
I mean xhrs that contain data (frames/samples) that can be parsed for a complete segment and not , for example 1.9 segments and then the user is waiting for the next download although the player already has most of the desired segment.

**Comment by cconcolato** [comment]
The application sets the segment duration, see the demo app slider called "Segment Duration (number of frames)". 

Segments don't exist in the original file. And they are not even needed by MSE, only fragments are. But since I generate (so far) a fragment per frame (!), and since it is too computationally expensive for the browser to process each fragment one by one, I group them in segments. This also allows saving the segments as files. So the segments are only created on the fly for ease of processing on the client side but indeed if the frames to form the whole segments are not there, the segment will not be flushed, so you need to make sure that you fetch all those frames (thus the 2nd call to seek). Is that clearer ?

**Comment by guypaskar** [comment]
I understand all you said I just don't know how to translate from the 
Segment Duration (number of frames) to something that is of type time (seconds) to make the second seek.

How do you translate the number of frames to a duration in time?

**Comment by guypaskar** [comment]
Yes. I think add would be better. That way one can choose how to define the segment duration.
